app/src/App.js

```diff
@@ -60,16 +60,28 @@ class App extends Component {
     request.open('POST', compileURL)
     request.setRequestHeader('Content-Type', 'application/json')
     request.addEventListener("load", (event) => {
-      const response = JSON.parse(event.target.responseText)
-      if (event.target.status == 200) {
-        console.log(response)
-        onCompileSucceeded(response)
-      } else {
-        onCompileFailed(response)
+      switch (request.status) {
+        case 0:
+          const response = JSON.parse(event.target.responseText)
+          if (event.target.status == 200) {
+            console.log(response)
+            onCompileSucceeded(response)
+          } else {
+            onCompileFailed(response)
+          }
+          break
+
+        case 404:
+          onCompileFailed({status: 'failed', message: `Vyper compiler not found at "${compileURL}".`})
+          break
+
+        default:
+          onCompileFailed({status: 'failed', message: `Unknown error has occured at "${compileURL}".`})
```

Comment:
How about showing the content of response?

Reply:
nice suggestion!